### ISSUE TITLE
Docker immediately closes after starting server

### ISSUE THREAD

**Issue by ihowell** [itzg/docker-minecraft-server#480]
I am running docker on my mac laptop, running OSX 10.15.3. I use the following to start the server:
```
docker run --name minecraft-server -d -v /Users/ihowell/minecraft-server:/data -p 25565:25565 -e EULA=TRUE -e MEMORY=4G itzg/minecraft-server
```

I keep pinging `docker ps` and it says that `itzg/minecraft-server` is starting, but eventually, it just stops altogether and I have no idea why. I also don't see anything odd in the logs:
```
$ docker logs -f minecraft-server
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x   15 1000     1000           480 Apr  4 16:55 /data'
[init] Resolved version given LATEST into 1.15.2
[init] Resolving type given VANILLA
[init] server.properties already created, skipping
[init] log4j2.xml already created, skipping
[init] Checking for JSON files.
[init] Copying any mods over...
[init] Setting initial memory to 4G and max to 4G
[init] Starting the Minecraft server...
[16:57:53] [main/WARN]: Ambiguity between arguments [teleport, destination] and [teleport, targets] with inputs: [Player, 0123, @e, dd12be42-52a9-4a91-a8a1-11c01849e498]
[16:57:53] [main/WARN]: Ambiguity between arguments [teleport, location] and [teleport, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[16:57:53] [main/WARN]: Ambiguity between arguments [teleport, location] and [teleport, targets] with inputs: [0.1 -0.5 .9, 0 0 0]
[16:57:53] [main/WARN]: Ambiguity between arguments [teleport, targets] and [teleport, destination] with inputs: [Player, 0123, dd12be42-52a9-4a91-a8a1-11c01849e498]
[16:57:53] [main/WARN]: Ambiguity between arguments [teleport, targets, location] and [teleport, targets, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[16:57:53] [Server thread/INFO]: Starting minecraft server version 1.15.2
[16:57:53] [Server thread/INFO]: Loading properties
[16:57:53] [Server thread/INFO]: Default game type: SURVIVAL
[16:57:53] [Server thread/INFO]: Generating keypair
[16:57:53] [Server thread/INFO]: Starting Minecraft server on *:25565
[16:57:53] [Server thread/INFO]: Using epoll channel type
[16:57:54] [Server thread/INFO]: Preparing level "world"
[16:57:54] [Server thread/INFO]: Found new data pack vanilla, loading it automatically
[16:57:54] [Server thread/INFO]: Reloading ResourceManager: Default
2020/04/04 16:58:06 Done
```

Any idea what is going on?

**Comment by itzg** [comment]
Strange, I don't see any obvious causes there. Can you try the default 1G for the `MEMORY` setting? Speaking of which, have you increased the Docker Desktop VM to something greater than 4G? I would have expected a different failure if it ran out of memory, but thought I'd check since the default is 2G for the VM.

**Comment by ihowell** [comment]
Oh, I think that did it! Removing the memory setting worked. Do you know how to increase the memory capacity in Docker?

**Comment by ihowell** [comment]
Edit: Odd stuff going on. If I start the server using the service: `sudo systemctl start minecraft-server`, then I get the following as output:
```
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x    6 1000     1000          4096 Apr  5 17:00 /data'
[init] Resolved version given LATEST into 1.15.2
[init] Resolving type given VANILLA
[init] server.properties already created, skipping
[init] log4j2.xml already created, skipping
[init] Checking for JSON files.
[init] Copying any mods over...
[init] Setting initial memory to 1G and max to 1G
[init] Starting the Minecraft server...
```

Then, if I stop the service: `sudo systemctl stop minecraft-server`, then the server actually starts and docker is not shutdown:

```
[17:03:09] [main/WARN]: Ambiguity between arguments [teleport, destination] and [teleport, targets] with inputs: [Player, 0123, @e, dd12be42-52a9-4a91-a8a1-11c01849e498]
[17:03:09] [main/WARN]: Ambiguity between arguments [teleport, location] and [teleport, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[17:03:09] [main/WARN]: Ambiguity between arguments [teleport, location] and [teleport, targets] with inputs: [0.1 -0.5 .9, 0 0 0]
[17:03:09] [main/WARN]: Ambiguity between arguments [teleport, targets] and [teleport, destination] with inputs: [Player, 0123, dd12be42-52a9-4a91-a8a1-11c01849e498]
[17:03:09] [main/WARN]: Ambiguity between arguments [teleport, targets, location] and [teleport, targets, destination] with inputs: [0.1 -0.5 .9, 0 0 0]
[17:03:09] [Server thread/INFO]: Starting minecraft server version 1.15.2
[17:03:09] [Server thread/INFO]: Loading properties
[17:03:09] [Server thread/INFO]: Default game type: SURVIVAL
[17:03:09] [Server thread/INFO]: Generating keypair
[17:03:09] [Server thread/INFO]: Starting Minecraft server on *:25565
[17:03:09] [Server thread/INFO]: Using epoll channel type
[17:03:10] [Server thread/INFO]: Preparing level "world"
[17:03:10] [Server thread/INFO]: Reloading ResourceManager: Default
[17:03:11] [Server thread/INFO]: Loaded 6 recipes
[17:03:12] [Server thread/INFO]: Loaded 825 advancements
[17:03:19] [Server thread/INFO]: Preparing start region for dimension minecraft:overworld
[17:03:20] [Server thread/INFO]: Preparing spawn area: 0%
[17:03:20] [Server thread/INFO]: Preparing spawn area: 0%
[17:03:20] [Server thread/INFO]: Preparing spawn area: 0%
[17:03:22] [Server-Worker-6/INFO]: Preparing spawn area: 83%
[17:03:22] [Server-Worker-7/INFO]: Preparing spawn area: 83%
[17:03:22] [Server-Worker-2/INFO]: Preparing spawn area: 83%
[17:03:22] [Server-Worker-7/INFO]: Preparing spawn area: 83%
[17:03:23] [Server-Worker-7/INFO]: Preparing spawn area: 91%
[17:03:23] [Server-Worker-1/INFO]: Preparing spawn area: 91%
[17:03:24] [Server-Worker-7/INFO]: Preparing spawn area: 91%
[17:03:24] [Server-Worker-2/INFO]: Preparing spawn area: 93%
[17:03:25] [Server-Worker-2/INFO]: Preparing spawn area: 95%
[17:03:25] [Server-Worker-3/INFO]: Preparing spawn area: 99%
[17:03:25] [Server thread/INFO]: Time elapsed: 6024 ms
[17:03:25] [Server thread/INFO]: Done (15.700s)! For help, type "help"
[17:03:25] [Server thread/INFO]: Starting remote control listener
[17:03:25] [RCON Listener #1/INFO]: RCON running on 0.0.0.0:25575
```

What is going on?

**Comment by itzg** [comment]
I'm glad to hear that was indeed the issue, but not sure what would be the cause on your Ubuntu system.

FYI, the VM memory can be increased in the Docker Desktop's Resources > Advanced settings, such as shown here

![image](https://user-images.githubusercontent.com/988985/78510761-4dd9ba80-775d-11ea-9dec-f16ed79d71fd.png)
